src/main/java/frc/robot/subsystems/flywheel/FlywheelIOCTRE.java

```diff
@@ -0,0 +1,112 @@
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Current;
+import edu.wpi.first.units.measure.Voltage;
+
+public class FlywheelIOCTRE implements FlywheelIO {
+  public static final double GEAR_RATIO = 1.5;
+
+  public final TalonFX leader = new TalonFX(14);
+  public final TalonFX follower = new TalonFX(15);
+
+  private final VoltageOut m_voltReq = new VoltageOut(0.0);
+
+  private final StatusSignal<Angle> leaderPosition = leader.getPosition();
+  private final StatusSignal<AngularVelocity> leaderVelocity = leader.getVelocity();
+  private final StatusSignal<Voltage> leaderAppliedVolts = leader.getMotorVoltage();
+  private final StatusSignal<Current> leaderStatorCurrent = leader.getStatorCurrent();
+  private final StatusSignal<Current> followerStatorCurrent = follower.getStatorCurrent();
+  private final StatusSignal<Current> leaderSupplyCurrent = leader.getSupplyCurrent();
+  private final StatusSignal<Current> followerSupplyCurrent = follower.getSupplyCurrent();
+
+  private final Debouncer leaderDebounce = new Debouncer(0.5);
+  private final Debouncer followerDebounce = new Debouncer(0.5);
+
+  public FlywheelIOCTRE() {
+    var config = new TalonFXConfiguration();
+    config.CurrentLimits.StatorCurrentLimit = 30.0;
+    config.CurrentLimits.StatorCurrentLimitEnable = true;
+    config.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+    config.Slot0.kP = 5;
+    config.Slot0.kI = 0;
+    config.Slot0.kD = 0;
+    leader.getConfigurator().apply(config);
+    follower.getConfigurator().apply(config);
+    follower.setControl(new Follower(leader.getDeviceID(), false));
+
+    BaseStatusSignal.setUpdateFrequencyForAll(
+        50.0,
+        leaderPosition,
+        leaderVelocity,
+        leaderAppliedVolts,
+        leaderStatorCurrent,
+        followerStatorCurrent,
+        leaderSupplyCurrent,
+        followerSupplyCurrent);
+  }
+
+  @Override
+  public void updateInputs(FlywheelIOInputs inputs) {
+    var leaderStatus =
+        BaseStatusSignal.refreshAll(
+            leaderPosition,
+            leaderVelocity,
+            leaderAppliedVolts,
+            leaderStatorCurrent,
+            leaderSupplyCurrent);
+
+    var followerStatus = BaseStatusSignal.refreshAll(followerStatorCurrent, followerSupplyCurrent);
+
+    inputs.leaderConnected = leaderDebounce.calculate(leaderStatus.isOK());
+    inputs.followerConnected = followerDebounce.calculate(followerStatus.isOK());
+    inputs.position = leaderPosition.getValue().div(GEAR_RATIO);
+    inputs.velocity = leaderVelocity.getValue().div(GEAR_RATIO);
+    inputs.appliedVoltage = leaderAppliedVolts.getValue();
+    inputs.leaderStatorCurrent = leaderStatorCurrent.getValue();
+    inputs.followerStatorCurrent = followerStatorCurrent.getValue();
+    inputs.leaderSupplyCurrent = leaderSupplyCurrent.getValue();
+    inputs.followerSupplyCurrent = followerSupplyCurrent.getValue();
+
+    leader.optimizeBusUtilization(4, 0.1);
```

Comment:
this is interesting.. why dont' we apply this to all subsystems? is this something that could be calculated at startup ? or somethign that should be provided in the RobotConfig.json ;)